src/test/java/net/infordata/em/TerminalClientTest.java

```diff
@@ -290,4 +282,182 @@ public void shouldSendExceptionToExceptionHandlerWhenLoginAndServerDownByLabel()
     exceptionWaiter.awaitException();
   }
 
+  @Test
+  public void shouldGetCorrectFieldsWhenGetFields() throws Exception {
+    awaitKeyboardUnlock();
+   List<TestField> actualFields = client.getFields().stream().map(TestField::new).collect(Collectors.toList());
+    assertEquals(buildExpectedFields(), actualFields);
+  }
+
+  private List<TestField> buildExpectedFields() {
+    List<TestField> testFieldList = new ArrayList<>();
+    testFieldList.add((new TestField(new TestField.FieldBuilder(FIELD_STRING_OUTPUT).withAttr(36).withDefaultColumn().withRow(5))));
```

Comment:
in general the builder pattern should look more like ` new TestField.FieldBuilder()......build()` and not as a new inside another. Please implement build method on builder and use that instead to make code a little shorter, better encapsulated, and mor obvious. Additionally, I think you have an unnecessary pair of parenthesis in this expression.